### ISSUE TITLE
FR: secrets store

### ISSUE THREAD

**Issue by jjh42** [tailscale/tailscale#12010]
### What are you trying to do?

Tailscale already authenticates/verifes all the machines on a network.

It seems then it would be kinda natural if tailscale could operate as a secrets store (e.g. some way to configure secrets and configuring which machines can access them via tags).

### What is the impact of not solving this?

one has to use some other secrets store.

**Comment by bradfitz** [comment]
See https://tailscale.com/blog/acl-grants which also describes https://github.com/tailscale/setec

That's what we use ourselves.

**Comment by jjh42** [comment]
thanks. will take a look.

I guess this FR could be considered: could tailscale have a managed version of setac automatically for my tailnet.

**Comment by jjh42** [comment]
is setec intend for use by people outside tailscale (looks fairly new, no docs about how to setup the server, github issues are disable)?

**Comment by creachadair** [comment]
Hi @jjh42,

Yes, setec is fairly new and is not an official product, but we use it internally at Tailscale and we made it public with the intent that others should be able to use it too.

You're right that it could use some more documentation, I appreciate the feedback. I just wrote up https://github.com/tailscale/setec/pull/110 with some basics on how to run a server. I'm sure there's more we could write, but I hope that will be enough to help you get started.

Having a Tailscale-hosted version can certainly remain a feature request, and I'll update the title to reflect that in a moment. To set expectations, however, I don't expect we are likely to work on a Tailscale-hosted secrets service in the near future.

**Comment by jjh42** [comment]
Thanks.

**Comment by jjh42** [comment]
Would you accept PRs on setec (e.g. to support alternative kms providers)?

**Comment by jjh42** [comment]
For issues/bugs anywhere we should report. setec seems to be running at 100% of CPU all the time (with no load)

**Comment by jjh42** [comment]
also is the ACL grants a supported setup in tailscale. The only documentation I see is the blog post. Wondering about using it for other things.

**Comment by creachadair** [comment]
> also is the ACL grants a supported setup in tailscale. The only documentation I see is the blog post. Wondering about using it for other things.

Yes, [ACL grants](https://tailscale.com/kb/1324/acl-grants) are also documented in the knowledge base. And yes, you should be able to introduce your own application capabilities. The app itself decides how to interpret the grant metadata so you can write it to do whatever you like.

> For issues/bugs anywhere we should report. setec seems to be running at 100% of CPU all the time (with no load)

For now let's keep issues here in this repository. As far as the symptom you're describing, can you share any other information about your setup? Are you running it based on the examples in my PR, and does the node you created show up in your tailnet? I tested those examples on my personal tailnet when I was writing up the doc, but it's possible I missed something.

**Comment by jjh42** [comment]
@creachadair thanks. yeah we are running it more as specified in the doc on AWS. Everything appears to be working fine just CPU usage of the setec server process is basically 100% all the time (and there is very little load on the test instance) which seems a little worrying.

**Comment by jjh42** [comment]
other than that its working great.

**Comment by creachadair** [comment]
> @creachadair thanks. yeah we are running it more as specified in the doc on AWS. Everything appears to be working fine just CPU usage of the setec server process is basically 100% all the time (and there is very little load on the test instance) which seems a little worrying.

That's interesting. I am not sure why you're seeing that, it is not expected (and does not occur for my installation). Are you able to see anything in the logs that might indicate what it's doing?
